packages/platform-sdk-egld/src/services/transaction.ts

```diff
@@ -42,18 +52,28 @@ export class TransactionService implements Contracts.TransactionService {
 		const unsignedTransaction = {
 			sender: input.from,
 			receiver: input.data.to,
-			value: new BigVal(input.data.amount, "coins"),
+			value: Balance.egld(input.data.amount),
 			gasPrice: (input.fee as unknown) as number,
 			gasLimit: (input.feeLimit as unknown) as number,
 			data: input.data.memo,
 		};
 
-		const signedTransaction = await BasicWallet.fromMnemonic(input.sign.mnemonic).signTransaction(
-			unsignedTransaction,
-			this.#provider,
-		);
-
-		return new SignedTransactionData(signedTransaction.signature, unsignedTransaction, signedTransaction);
+		const mnemonic: Mnemonic = Mnemonic.fromString(input.sign.mnemonic);
+		const secretKey: UserSecretKey = mnemonic.deriveKey(0);
+		const signer: UserSigner = new UserSigner(secretKey);
+
+		const tx: Transaction = new Transaction(
+			{
+				receiver: Address.fromString(input.data.to),
+				value: Balance.egld(input.data.amount),
+				gasPrice: new GasPrice((input.fee as unknown) as number),
+				gasLimit: new GasLimit((input.feeLimit as unknown) as number),
+				data: new TransactionPayload(input.data.memo),
+			});
+		// tx.setNonce(account.nonce); // TODO Do we need specify this? If so we will need to sync account first, or receive it via param
```

Comment:
Unless the library automatically sets the nonce we will need to pass it in.

Reply:
@marianogoldman but we don't need to sync here, it's parameter that can already be passed in.

Reply:
I agree... this is their [example code](https://elrondnetwork.github.io/elrond-sdk-docs/erdjs/latest/#creating-value-transfer-transactions) for signing transaction.

I think we need to pass it in, as I don't think the library has the means (at this stage) to figure that out. And yes, I guess it should come as a param in the `TransactionInputs`.

Reply:
We don't have any tests that actually ever submit a transaction (like we have for Cardano). Has this ever been tested against the network?

Reply:
I've manually tested it with the previous lib.